Src/HALALMock/Services/Communication/Ethernet/TCP/ServerSocket.cpp

```diff
@@ -163,7 +141,128 @@ void ServerSocket::send(){
 bool ServerSocket::is_connected(){
 	return state == ServerSocket::ServerState::ACCEPTED;
 }
+ServerSocket::create_server_socket(){
+	server_socket_fd = socket(AF_INET, SOCK_STREAM, 0);
+	if(server_socket_fd == -1){
+		std::cout<<"Socket creation failure\n";
+		return;
+	}
+	//inset the local address and port
+	struct sockaddr_in server_socket_Address;
+	server_socket_Address.sin_family = AF_INET;
+	server_socket_Address.sin_addr.s_addr = local_ip.address;
+	server_socket_Address.sin_port = htons(local_port);
+	if(bind(server_socket_fd, (struct sockaddr*)&server_socket_Address, sizeof(server_socket_Address)) < 0){
+		std::cout<<"Bind error\n";
+		close(server_socket_fd);
+		return;
+	}
+}
+ServerSocket::configure_server_socket(){
+	//to reuse local address:
+	int opt = 1;
+	if (setsockopt(server_socket_fd, SOL_SOCKET, SO_REUSEADDR, &opt, sizeof(opt)) < 0) {
+    	std::cerr << "Error setting SO_REUSEADDR\n";
+   		close(server_socket_fd);
+    	return;
+	}
+	//disable naggle algorithm
+	int flag = 1;
+	if (setsockopt(server_socket_fd,IPPROTO_TCP,TCP_NODELAY,(char *) &flag, sizeof(int)) < 0){
+		std::cout<<"It has been an error disabling Nagle's algorithm\n";
+		close(server_socket_fd);
+		return;
+	}
+	//habilitate keepalives
+    int optval = 1;
+    if (setsockopt(server_socket_fd, SOL_SOCKET, SO_KEEPALIVE, &optval, sizeof(optval)) < 0) {
+        std::cout << "ERROR configuring KEEPALIVES\n";
+        close(server_socket_fd);
+        return;
+    }
+	// Configurar TCP_KEEPIDLE it sets what time to wait to start sending keepalives 
+    float tcp_keepidle_time = static_cast<float>(keepalive_config.inactivity_time_until_keepalive_ms)/1000.0;
+    if (setsockopt(server_socket_fd, IPPROTO_TCP, TCP_KEEPIDLE, &tcp_keepidle_time, sizeof(tcp_keepidle_time)) < 0) {
+        std::cout << "Error configuring TCP_KEEPIDLE\n";
+		close(server_socket_fd);
+        return;
+    }
+	  //interval between keepalives
+    float keep_interval_time = static_cast<float>(keepalive_config.space_between_tries_ms)/1000.0;
+    if (setsockopt(server_socket_fd, IPPROTO_TCP, TCP_KEEPINTVL, &keep_interval_time, sizeof(keep_interval_time)) < 0) {
+        std::cout << "Error configuring TCP_KEEPINTVL\n";
+        close(server_socket_fd);
+        return;
+    }
+	 // Configure TCP_KEEPCNT (number keepalives are send before considering the connection down)
+	 float keep_cnt = static_cast<float>(keepalive_config.tries_until_disconnection)/1000.0;
+    if (setsockopt(server_socket_fd, IPPROTO_TCP, TCP_KEEPCNT, &keep_cnt, sizeof(keep_cnt)) < 0) {
+        std::cout << "Error to configure TCP_KEEPCNT\n";
+        close(server_socket_fd);
+        return ;
+    }
```

Comment:
I would still separate into more functions, but just a nitpick, also, return a `bool` that is `true` if the configuration was successful or `false` in any other case